File: Problems_and_Solutions/days-10 to 19/day-12/Java/Solution1.java
import java.util.Arrays;
class Solution1 {
    public static void insertionSort(int[] arr, int n) {
        if (n == 0) return;

        insertionSort(arr,n-1);

        for(int j=n; j>0; j--){
            if(arr[j-1] > arr[j]){
                int temp = arr[j-1];
                arr[j-1] = arr[j];
                arr[j] = temp;
            }
        }
    }

    public static void main(String[] args){
        int a[] = {4, 1, 3, 9, 7};
        insertionSort(a,a.length-1);
        System.out.println(Arrays.toString(a));
    }
}

File: Problems_and_Solutions/days-10 to 19/day-12/c++/solution 1.cpp
class Solution {
  public:
    // Please change the array in-place
    void insertionSort(vector<int>& arr) {
        // code here
       rec1(arr,1);
    }
    void rec1(vector<int>& arr,int i){
        if(i<arr.size())
        {
            rec2(arr,i);
            rec1(arr,i+1);
        }
        
    }
    void rec2(vector<int>& arr,int j){
        if(j>0)
        {
            if(arr[j]<arr[j-1])
            {
                int temp=arr[j];
                arr[j]=arr[j-1];
                arr[j-1]=temp;
                rec2(arr,j-1);
            }
            else
                return;
            
        }
    }
};